client/client.go

```diff
@@ -18,6 +18,8 @@ type Client interface {
 	BulkIndexClose(context.Context) error
 	Checker(ctx context.Context, state *health.CheckState) error
 	CreateIndex(ctx context.Context, indexName string, indexSettings []byte) error
+	DeleteDocumentByID(ctx context.Context, index, documentID string) error
```

Comment:
Supplying an ID when deleting a document is the implied default so the suffix is redundant. 
```suggestion
	DeleteDocument(ctx context.Context, index, documentID string) error
```